homeassistant/components/mqtt/server.py

```diff
@@ -48,29 +37,22 @@ def start(hass, server_config):
         else:
             client_config = None
 
-        start_server = asyncio.gather(broker_coro(loop, server_config),
-                                      loop=loop)
-        loop.run_until_complete(start_server)
-        # Result raises exception if one was raised during startup
-        broker = start_server.result()[0]
+        future = run_coroutine_threadsafe(
+            broker_coro(hass.loop, server_config), hass.loop)
+
+        broker = future.result()
```

Comment:
It is a bit unclear what happens now when an exception is raised during start of the server. The exception is printed to the console but it is not raised by `future.result()` as I would expect.

To get it to raise an exception locally:

```yaml
mqtt:
  embedded: {}
```